### PR TITLE
tutorials: add circle ci snap tutorial

### PR DESCRIPTION
**Pull request by kyrofa**
This uses some new functionality in Snapcraft v2.37 to support other CI systems. It re-uses as much as possible from the Travis CI tutorial.

### CODE REVIEW

**Comment by caldav** [comment]
Demo started here: http://tutorials.ubuntu.com-pr-618.run.demo.haus/tutorial/continuous-snap-delivery-from-circle-ci

File: tutorials/packaging/continuous-snap-delivery-from-circle-ci/continuous-snap-delivery-from-circle-ci.md [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+snapcraft login
+snapcraft register hello-yourname
+```
+
+We now have a GitHub project containing packaging details for a snap that has been registered to our account. Let's move onto Circle CI.
+
+## Setting up Circle CI
+Duration: 5:00
+
+To use Circle CI, first we will have to sign up there. Let's go to [https://circleci.com/vcs-authorize] and click "Log In with GitHub". Then, click "Authorize application" to allow Circle CI to access your GitHub account.
```
Comment (caldav) [inline tutorials/packaging/continuous-snap-delivery-from-circle-ci/continuous-snap-delivery-from-circle-ci.md]: Link formatting needs fixing

File: tutorials/packaging/continuous-snap-delivery-from-circle-ci/continuous-snap-delivery-from-circle-ci.md [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+
+Now, you should see a list of GitHub repositories, including your snap. If you don't see this list, there should be an "Add projects" button. Enable the GitHub repository of your snap by clicking the "Setup project" button next to it.
+
+![](images/setup_project.png)
+
+Once you select that button, you'll be given a choice between the operating system and platform you'd like to use. You'll need to use Linux (of course), and while the platform is up to you we recommend version 2.0, and will use that here.
+
+You can even select a language, although since we're only building snaps here, we'll just use "Other".
+
+Finally, it gives you a suggestion for a basic `.circleci/config.yml`. No need to do any of that, just hit "Start building". That will set things up such that Circle CI will be triggered whenever you make changes to your repository. You'll notice it start to build the current state of the repository, but will soon realize there is nothing to run (because we haven't done that, yet).
```
Comment (caldav) [inline tutorials/packaging/continuous-snap-delivery-from-circle-ci/continuous-snap-delivery-from-circle-ci.md]: notice it start**s** to build

File: tutorials/packaging/continuous-snap-delivery-from-circle-ci/continuous-snap-delivery-from-circle-ci.md [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+  - The [Circle CI 2.0 docs] have a lot of information about adding tests and other things to your CI executions.
+
+[Create your first snap]: /tutorial/create-your-first-snap
+[https://github.com]: https://github.com/
+[https://github.com/new]: https://github.com/new
+[https://dashboard.snapcraft.io]: https://dashboard.snapcraft.io/
+[four channels]: https://snapcraft.io/docs/reference/channels#channels
+[this demo repository]: https://github.com/kyrofa/hello-snap-kyrofa
+[use the snapcraft build service]: https://build.snapcraft.io/
+[tutorials]: https://tutorials.ubuntu.com
```
Comment (caldav) [inline tutorials/packaging/continuous-snap-delivery-from-circle-ci/continuous-snap-delivery-from-circle-ci.md]: This one could be changed to: https://tutorials.ubuntu.com?q=snap

File: tutorials/packaging/continuous-snap-delivery-from-circle-ci/continuous-snap-delivery-from-circle-ci.md (hunk 1 above)
Comment (kyrofa) [reply tutorials/packaging/continuous-snap-delivery-from-circle-ci/continuous-snap-delivery-from-circle-ci.md]: Dang, I knew I was gonna miss at least one! Thank you-- fixed.

File: tutorials/packaging/continuous-snap-delivery-from-circle-ci/continuous-snap-delivery-from-circle-ci.md (hunk 2 above)
Comment (kyrofa) [reply tutorials/packaging/continuous-snap-delivery-from-circle-ci/continuous-snap-delivery-from-circle-ci.md]: Good catch, fixed.

File: tutorials/packaging/continuous-snap-delivery-from-circle-ci/continuous-snap-delivery-from-circle-ci.md (hunk 3 above)
Comment (kyrofa) [reply tutorials/packaging/continuous-snap-delivery-from-circle-ci/continuous-snap-delivery-from-circle-ci.md]: Very good, fixed.

**Pull request merged**